# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/layers/client/TextureClient.cpp
# Commit: a4031c701928
# Full Hash: a4031c7019280da3be78ab791aa7bbb123ec8c02
# Author: Jamie Nicol <jnicol@mozilla.com>
# Date: 2024-07-23 09:34:47
# Regressor Bug: 1898238
# File Overlap Count: 1
# Description:
#   Bug 1898238 - Use SharedSurface_EGLImage for out-of-process webgl and accelerated canvas on Android. r=sotaro,jgilbert
#   
#   Adds a pref wegb.enable-egl-image which is true by default. When
#   enabled, we use EGL images as the preferred texture type for
#   out-of-process webgl and accelerated canvas on Android. In-process
# ==============================================================================

diff -r 7e0171e88041 -r a4031c701928 gfx/layers/client/TextureClient.cpp
--- a/gfx/layers/client/TextureClient.cpp	Mon Jul 22 20:23:38 2024 +0000
+++ b/gfx/layers/client/TextureClient.cpp	Mon Jul 22 20:23:39 2024 +0000
@@ -375,7 +375,8 @@
   if (aAllocFlags & ALLOC_FORCE_REMOTE) {
     RefPtr<CanvasChild> canvasChild = aAllocator->GetCanvasChild();
     if (canvasChild) {
-      TextureType webglTextureType = TexTypeForWebgl(aKnowsCompositor);
+      TextureType webglTextureType =
+          TexTypeForWebgl(aKnowsCompositor, /* aIsWebglOop */ true);
       if (canvasChild->EnsureRecorder(aSize, aFormat, textureType,
                                       webglTextureType)) {
         return new RecordedTextureData(canvasChild.forget(), aSize, aFormat,